docs/tutorial/composition.md

```diff
@@ -0,0 +1,219 @@
+|     6     || 34        | 34        | 55        | 55        |
+|     7     || 55        | 55        | 89        | 89        |
+|     8     || 89        | 89        | 144       | 144       |
+|     9     || 144       | 144       | 233       | 233       |
+```
+
+Though it is not visible in this example, you don't need to have a matching number of inputs and outputs. Nor do you need to have a specific logic in mind for how you forward values from outputs to inputs; those can be switched for instance. 
+
+{% hint style="info" %}
+If you wonder why we wrapped the outputs into `identity`, see below in the [Gotcha section](#gotcha).
+{% endhint %}
+
+## Iteration support
+
+In the previous example you can see that to some extent, composition allows to support iteartion with cleartext iterands. That is loops with shape :
+
+```python
+for i in some_cleartext_range:
+    # Do something in FHE in the loop body, implement as an fhe circuit.
+```
+
+With this pattern, we can also support while loops with dynamic condition, as long as this condition is computed in cleartext. Here is an example that computes the [Collatz sequence](https://en.wikipedia.org/wiki/Collatz_conjecture):
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"x": "encrypted"})
+def collatz(x):
+
+    y = x // 2
+    z = 3 * x + 1
+
+    is_x_odd = fhe.bits(x)[0]
+
+    # In a fast way, compute ans = is_x_odd * (z - y) + y
+    ans = fhe.multivariate(lambda b, x: b * x)(is_x_odd, z - y) + y
+
+    return ans
+
+
+print("Compiling `collatz` function ...")
+inputset = [i for i in range(63)]
+collatz_fhe = collatz.compile(inputset, composable=True)
+
+print("Generating keyset ...")
+collatz_fhe.keygen()
+
+print("Encrypting initial value")
+x = 19
+x_enc = collatz.encrypt(x)
+result = None
+
+print(f"| iteration || decrypted | cleartext |")
+while collatz.decrypt(x_enc) != 1:
+    x_enc = collatz_fhe.run(x_enc)
+    x = collatz(x)
+
+    x_dec = collatz_fhe.decrypt(x_enc)
+    print(f"|     {i}     || {x_dec:<9} | {x:<9} |")
+```
+
+Which should print:
+
+```shell
+Compiling `collatz` function ...
+Generating keyset ...
+Encrypting initial value
+| decrypted | cleartext |
+| 58        | 58        |
+| 29        | 29        |
+| 88        | 88        |
+| 44        | 44        |
+| 22        | 22        |
+| 11        | 11        |
+| 34        | 34        |
+| 17        | 17        |
+| 52        | 52        |
+| 26        | 26        |
+| 13        | 13        |
+| 40        | 40        |
+| 20        | 20        |
+| 10        | 10        |
+| 5         | 5         |
+| 16        | 16        |
+| 8         | 8         |
+| 4         | 4         |
+| 2         | 2         |
+| 1         | 1         |
+```
+
+Here we use a while loop that keeps iterating as long as the decryption of the running value is different from `1`. Again, the loop body is implemented in fhe, but the iteration control has to be in cleartext.
+
+## Gotcha
```

Comment:
Maybe something about precision unification and crypto params ?
e.g.   ((15 * x) // 16) * 15  is more costly with composable
e.g.  two input, 1 TLU for each,  1 is 1 bit, the other is 8bit, everything is now 8bit which is costlier

Reply:
I would say it is worth mentioning the general reasons why performances are affected, but maybe not go into so many details. I tried something, tell me what you think about it.